### DISCUSSION TITLE
Micropyton for SAM9X60D1G-I/LZB

### DISCUSSION THREAD

**Discussion by networker-x** [micropython/micropython#15839] [Ideas]
Being the not-so-bright software engineer, mainly because I am a hardware engineer, I figured that micropython would be perfect for the Microchip SAM9X60D1G-I/lZB system on module, The real question is who will make it possible ? 

DigiKey:https://www.digikey.bg/en/products/detail/microchip-technology/sam9x60d1g-i-lzb/15985687

[SAM9X60-Data-Sheet-DS60001579E.pdf](https://github.com/user-attachments/files/16978992/SAM9X60-Data-Sheet-DS60001579E.pdf)
![MFG_SAM9X60D1G-ILZB](https://github.com/user-attachments/assets/5907a41e-5b9f-4bc0-8d32-c6d26bee0196)

**Comment by bidrohini** [comment]
MicroPython already has support for various ARM Cortex-M processors, but the ARM9 architecture of the SAM9X60 might require custom adaptations.

**Comment by robert-hh** [comment]
Adapting the clock system and peripherals is the major task. And it requires the availability of development boards with a good (=low) price tag, like STM does. 
From the first glance this MCU plays in the league of a MIMXRT106x, used in the Teensy 4.x board. For these there is already a MP ports present.
Edit: With the performance figures of this module I would not be surprised if already a Linux port is available.
Edit2: Found them: https://www.linux4sam.org/bin/view/Linux4SAM/Sam9x60CuriosityMainPage#Demo
